src/db/collections/analytics/daily_active_addresses.rs

```diff
@@ -0,0 +1,98 @@
+};
+
+/// Computes the activity of addresses within a milestone.
+#[derive(Debug, Default)]
+pub struct DailyActiveAddressesAnalytics {
+    current_date: Option<Date>,
+}
+
+#[derive(Copy, Clone, Debug, Default, PartialEq, Eq, Serialize, Deserialize)]
+pub struct DailyActiveAddressAnalyticsResult {
+    pub count: u64,
+}
+
+#[async_trait]
+impl Analytic for DailyActiveAddressesAnalytics {
+    async fn get_measurement(
+        &mut self,
+        db: &MongoDb,
+        _: MilestoneIndex,
+        milestone_timestamp: MilestoneTimestamp,
+    ) -> Result<Option<Measurement>, Error> {
+        let incoming_date = OffsetDateTime::try_from(milestone_timestamp)?.date();
+
+        if self.current_date.is_none() {
+            self.current_date = Some(incoming_date);
+        }
+
+        if let Some(current_date) = self.current_date {
+            if current_date.next_day() == Some(incoming_date) {
+                let from = current_date.midnight().assume_utc();
+                let to_exclusive = incoming_date.midnight().assume_utc();
+
+                let res = db
+                    .collection::<OutputCollection>()
+                    .get_interval_address_activity_analytics(from.into(), to_exclusive.into())
+                    .await
+                    .map(|measurement| {
+                        Some(Measurement::DailyActiveAddressAnalytics(TimeInterval {
+                            from,
+                            to_exclusive,
+                            inner: measurement,
+                        }))
+                    });
+
+                self.current_date = Some(incoming_date);
+                return res;
+            }
+        }
+
+        Ok(None)
+    }
+}
+
+impl OutputCollection {
+    /// Create active address statistics for a given time interval `[from,to)`.
+    #[tracing::instrument(skip(self), err, level = "trace")]
+    pub async fn get_interval_address_activity_analytics(
+        &self,
+        from: MilestoneTimestamp,
+        to: MilestoneTimestamp,
+    ) -> Result<DailyActiveAddressAnalyticsResult, Error> {
+        Ok(self
+            .aggregate(
+                vec![
+                    doc! { "$match": doc! { "$or": [
+                        doc! { "$and": [
+                            doc! { "metadata.booked.milestone_timestamp": doc! { "$gte": from } },
+                            doc! { "metadata.booked.milestone_timestamp": doc! { "$lt": to } },
+                        ] },
+                        doc! { "$and": [
+                            doc! { "metadata.spent_metadata.spent.milestone_timestamp": doc! { "$gte": from } },
+                            doc! { "metadata.spent_metadata.spent.milestone_timestamp": doc! { "$lt": to } }
+                        ] },
+                    ] } },
+                    doc! { "$group": doc! { "_id": doc! { "addr": "$details.address" } } },
+                    doc! { "$count": "count" },
```

Comment:
I know that's copied from Compass, but I would still remove the redundant `doc!` macro calls.